test/engine/controller/logtail_control.go

```diff
@@ -36,6 +40,31 @@ func (l *LogtailController) Init(parent *CancelChain, fullCfg *config.Case) erro
 	// todo currently only support one fixed config with cgo
 	if len(fullCfg.Ilogtail.Config) == 1 {
 		cfg := fullCfg.Ilogtail.Config[0]
+		if len(cfg.Detail) != 0 {
+			_ = os.Remove(config.ConfigYamlFileDir)
+			_ = os.Mkdir(config.ConfigYamlFileDir, 0750)
+			for idx, detail := range cfg.Detail {
+				name := cfg.Name + "_" + strconv.Itoa(idx) + ".yaml"
+				if _, ok := detail["inputs"]; !ok {
+					return fmt.Errorf("lack of input plugin in the %d config detail under name %s", idx, cfg.Name)
+				}
+				if _, ok := detail["outputs"]; !ok {
+					detail["flushers"] = []map[string]interface{}{
+						{
+							"Type":    "flusher_grpc",
+							"Address": "host.docker.internal:8000",
+						},
+					}
```

Comment:
currently, only support yaml in pure plugin mode?

Reply:
yaml is now available for both file and plugin mode, see #210